### ISSUE TITLE
Validation Failure - yesod and yesod-mongodb-raw

### ISSUE THREAD

**Issue by hamiltont** [TechEmpower/FrameworkBenchmarks#1135]
Hitting `http://127.0.0.1:8000/dbs/foo` should detect that foo is not a number and choose to return 1 instead, as though we had queried `http://127.0.0.1:8000/dbs/1`. Instead it treats this as a path and returns 404

**Comment by LadyMozzarella** [comment]
yesod-mongodb-raw error snippet from round 10, preview 3 (yesod passing): 

```bash 
<html><head><title>Internal Server Error</title></head><body><h1>Internal Server Error</h1>
<pre>Pattern match failure in do expression at src/yesod.hs:157:5-10</pre>
```

**Comment by LadyMozzarella** [comment]
Closing. Fixed in master by #1440
